crates/ruff/src/rules/pydocstyle/rules/non_imperative_mood.rs

```diff
@@ -7,22 +9,34 @@ use crate::ast::types::Range;
 use crate::checkers::ast::Checker;
 use crate::docstrings::definition::{DefinitionKind, Docstring};
 use crate::registry::Diagnostic;
-use crate::rules::pydocstyle::helpers::normalize_word;
+use crate::rules::pydocstyle::{self, helpers::normalize_word};
 use crate::violation::Violation;
 use crate::visibility::{is_property, is_test};
 
 static MOOD: Lazy<Mood> = Lazy::new(Mood::new);
 
 /// D401
-pub fn non_imperative_mood(checker: &mut Checker, docstring: &Docstring) {
+pub fn non_imperative_mood(
+    checker: &mut Checker,
+    docstring: &Docstring,
+    property_decorators: &BTreeSet<String>,
+) {
     let (
         DefinitionKind::Function(parent)
         | DefinitionKind::NestedFunction(parent)
         | DefinitionKind::Method(parent)
     ) = &docstring.kind else {
         return;
     };
-    if is_test(cast::name(parent)) || is_property(checker, cast::decorator_list(parent)) {
+
+    if is_test(cast::name(parent))
+        || is_property(checker, cast::decorator_list(parent))
+        || pydocstyle::helpers::is_property(
```

Comment:
I think instead, maybe we should just pass the extra `property_decorators` to `is_property`. It's not super consistent with the other utilities in that file, which don't accept "extra" decorators, but it does mean one centralized function for determining "decorator status", _and_ it's more efficient since we only do the `checker.resolve_call_path` once per decorator.

Reply:
Made it like so